core/domain/user_services_test.py

```diff
@@ -1329,6 +1614,275 @@ def test_record_user_started_state_translation_tutorial(self):
+        created_exp_ids = ['exp1', 'exp2', 'exp3']
+        edited_exp_ids = ['exp2', 'exp3', 'exp4']
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        pre_add_contributions = user_services.get_user_contributions(user_id)
+
+        self.assertEqual(
+            [],
+            pre_add_contributions.created_exploration_ids)
+
+        self.assertEqual(
+            [],
+            pre_add_contributions.edited_exploration_ids)
+
+        user_services.update_user_contributions(
+            user_id,
+            created_exp_ids,
+            edited_exp_ids)
+
+        contributions = user_services.get_user_contributions(user_id)
+
+        self.assertEqual(
+            ['exp1', 'exp2', 'exp3'],
+            contributions.created_exploration_ids)
+
+        self.assertEqual(
+            ['exp2', 'exp3', 'exp4'],
+            contributions.edited_exploration_ids)
+
+    def test_update_user_contributions_for_invalid_user_raises_error(self):
+        non_existent_user_id = 'id_x'
+        created_exp_ids = ['exp1', 'exp2', 'exp3']
+        edited_exp_ids = ['exp2', 'exp3', 'exp4']
+        error_msg = (
+            'User contributions model for user %s does not exist.'
+            % non_existent_user_id)
+
+        with self.assertRaisesRegex(Exception, error_msg):
+            user_services.update_user_contributions(
+            non_existent_user_id,
+            created_exp_ids,
+            edited_exp_ids)
+
+    def test_add_created_exploration_id(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        created_exp_ids = ['exp1']
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        user_services.add_created_exploration_id(user_id, 'exp1')
+
+        contributions = user_services.get_user_contributions(user_id)
+
+        self.assertEqual(
+            created_exp_ids,
+            contributions.created_exploration_ids)
+
+    def test_add_created_exploration_id_creates_user_contribution(self):
+        user_id = 'id_x'
+        created_exp_id = 'exp1'
+
+        pre_add_contributions = user_services.get_user_contributions(user_id)
+        self.assertIsNone(pre_add_contributions)
+
+        user_services.add_created_exploration_id(user_id, created_exp_id)
+        contributions = user_services.get_user_contributions(user_id)
+
+        self.assertIsInstance(contributions, user_domain.UserContributions)
+        self.assertEqual(
+            ['exp1'],
+            contributions.created_exploration_ids)
+
+    def test_add_edited_exploration_id(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        edited_exp_ids = ['exp1']
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        user_services.add_edited_exploration_id(user_id, 'exp1')
+
+        contributions = user_services.get_user_contributions(user_id)
+
+        self.assertEqual(
```

Comment:
Missing pre-check.

Please check all your tests for this.

Reply:
I added pre-checks where applicable.